src/coreclr/vm/appdomain.cpp

```diff
@@ -2672,7 +2700,7 @@ void AppDomain::TryIncrementalLoad(Assembly *pAssembly, FileLoadLevel workLevel,
 
         //We will cache this error and wire this load to forever fail,
         // unless the exception is transient or the file is loaded OK but just cannot execute
-        if (!pEx->IsTransient() && !pAssembly->IsLoaded())
+        if (pAssembly != nullptr && !pEx->IsTransient() && !pAssembly->IsLoaded())
```

Comment:
I think I see why `pAssembly` can be `nullptr`. Can we get an assert in the `if (workLevel == FILE_LOAD_ALLOCATE)` clause that it can be be `nullptr`? I assume that is the logical flow in question. If it _might be_ `nullptr`, then is there another assert that can be added at that point?